src/library/scala/util/Random.scala

```diff
@@ -43,11 +44,27 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    */
   def nextDouble(): Double = self.nextDouble()
 
+  /** Returns the next pseudorandom, uniformly distributed double value
+   *  between min (inclusive) and max (exclusive) from this random number generator's sequence.
+   */
+  def between(minInclusive: Double, maxExclusive: Double): Double = ThreadLocalRandom.current().nextDouble(minInclusive, maxExclusive)
```

Comment:
Using `ThreadLocalRandom.current()` ignores the seed value, thus making it impossible to get reproducible sequences of values. Since `scala.util.Random` can be constructed with a specific seed value, I would expect all methods to honor it (and the scaladoc comment even states it explicitly).

We could replace the wrapped `java.util.Random` by a `java.util.SplittableRandom` (Java 8+ only). Unfortunately, `scala.util.Random` exposes the wrapped `self` value and for some reason `SplittableRandom` is not a subclass of `Random`. (`ThreadLocalRandom` OTOH is a subclass of `Random` but cannot be instantiated).

The backwards-compatible solution would be to reimplement all of the `between` methods the same way it is done for the `Float` version. Not particularly elegant but the implementaion is simple enough after all. Or we bite the bullet and break backwards compatibility by switching to `SplittableRandom`. Opinions?